cpp/include/raft/core/device_resources_manager.hpp

```diff
@@ -0,0 +1,609 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
```

Comment:
The implementation looks good. Can you also explicit add this to the docs (might be easiest to create a doxygen group and reference the group in the docs).